io/net/ethtool_test.py

```diff
@@ -102,19 +110,41 @@ def test_ethtool(self):
                               ignore_status=True)
             if ret.exit_status != 0:
                 self.fail("failed")
-        if not self.ping_check('-c 5'):
-            self.fail("ping failed after interface is up")
+        if not HostInfo.ping_check(self, self.iface, self.peer, '10000',
+                                   flood=True):
+            self.fail("flood ping test failed")
 
-    def ping_check(self, options):
+    def test_ethtool_toggle_priv_flags(self):
         '''
-        Checks if ping to peer works fine and returns True.
-        Returns False otherwise.
+        Toggle the priv flag settings of the driver.
         '''
-        cmd = "ping -I %s %s %s" % (self.iface, options, self.peer)
-        if process.system(cmd, shell=True, verbose=True,
-                          ignore_status=True) != 0:
-            return False
-        return True
+        priv_pass = []
+        priv_fail = []
+        cmd = "ethtool --show-priv-flags %s" % (self.iface)
+        ret = process.run(cmd, shell=True, verbose=True,
+                          ignore_status=True)
+        if ret.exit_status:
+            self.cancel("Device Doesn't support Private flags")
+        for line in ret.stdout.decode("utf-8").splitlines():
+            if "off" in line:
+                val = "on"
+            else:
+                val = "off"
+            if "flags" not in line:
+                priv_flag = line.split(':')[0]
+                cmd = "ethtool --set-priv-flags %s \"%s\" %s" % \
+                      (self.iface, priv_flag.rstrip(), val)
+                ret1 = process.run(cmd, shell=True, verbose=True,
+                                   ignore_status=True)
+                if ret1.exit_status:
+                    priv_fail.append(priv_flag.rstrip())
+                else:
+                    priv_pass.append(priv_flag.rstrip())
```

Comment:
@manvanthar I assume what you want to do is:
```
for every private flag:
    store current value - on/off
    try toggle the value.
    if pass:
        toggle again - to restore to original state
    if fails:
        append it to fail_list
        continue
if fail_list:
    print list of failures
```

Reply:
@narasimhan-v 

Here is the sampleout of show-priv-flags

ethtool --show-priv-flags eth4
Private flags for eth4:
rx_cqe_moder       : off
tx_cqe_moder       : off
rx_cqe_compress    : off
rx_striding_rq     : off
rx_no_csum_complete: off
xdp_tx_mpwqe       : off
sniffer            : off
dropless_rq        : off
per_channel_stats  : on
tx_xdp_hw_checksum : off

Logic is
1. ignore first line and toggle all the priv flags for the device
2. make a list of pass and failed priv flags
3. Ping_check with the overall toggled list.
4. Print the fail list.

Reply:
@manvanthar I get this. Shouldn't the test test toggling back ?
Also, shouldn't the test bring the flags back to original state ?